### ISSUE TITLE
[bug]: renaming trunk to long name returns SQL error and deletes trunk

### ISSUE THREAD

**Issue by gruubar** [FreePBX/issue-tracker#542]
### FreePBX Version

FreePBX 17

### Issue Description

I had an old, unused trunk that had a poorly descriptive name so I renamed it to something longer that explained what it was and got an "SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'name' at row 1" error page. When I went back to the trunks list that trunk is gone. I don't need it back as it was unused and disabled ( I didn't want to delete it 'just in case' ) but now it is gone. I made a new test trunk and tried the same thing and got the same result. I looked in the asterisk DB and it is not there at all. It's not a huge deal for me but it seemed like it could be a problem if it were a production trunk that happened to disappear like this. The log output is from the web error page when I made the test trunk and renamed it.

### Operating Environment

Freepbx 17.0.19.16
Asterisk 20.9.3
Debian 12

+---------------------+------------+-----------------------------------+-------------+-----------+
| Module              | Version    | Status                            | License     | Signature |
+---------------------+------------+-----------------------------------+-------------+-----------+
| accountcodepreserve |            | Not Installed (Locally available) | GPLv2       | Sangoma   |
| adv_recovery        |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| allowlist           | 17.0.1.1   | Enabled                           | GPLv3+      | Sangoma   |
| amd                 | 17.0.1     | Enabled                           | GPLv3+      | Sangoma   |
| announcement        | 17.0.2.1   | Enabled                           | GPLv3+      | Sangoma   |
| api                 | 17.0.1.6   | Enabled                           | AGPLv3+     | Sangoma   |
| areminder           |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| arimanager          | 17.0.1.1   | Enabled                           | GPLv3+      | Sangoma   |
| asterisk-cli        | 17.0.2     | Enabled                           | GPLv3+      | Sangoma   |
| asteriskinfo        | 17.0.2     | Enabled                           | GPLv3+      | Sangoma   |
| backup              | 17.0.5.61  | Enabled                           | GPLv3+      | Sangoma   |
| blacklist           | 17.0.1.2   | Enabled                           | GPLv3+      | Sangoma   |
| broadcast           |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| builtin             |            | Enabled                           |             | Unsigned  |
| bulkhandler         | 17.0.5     | Enabled                           | GPLv3+      | Sangoma   |
| calendar            | 17.0.4.20  | Enabled                           | GPLv3+      | Sangoma   |
| callaccounting      |            | Not Installed (Locally available) | Commercial+ | Sangoma   |
| callback            | 17.0.2.1   | Enabled                           | GPLv3+      | Sangoma   |
| callerid            |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| callforward         | 17.0.1.6   | Enabled                           | AGPLv3+     | Sangoma   |
| calllimit           |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| callrecording       | 17.0.3.8   | Enabled                           | AGPLv3+     | Sangoma   |
| callwaiting         | 17.0.3.4   | Enabled                           | GPLv3+      | Sangoma   |
| cdr                 | 17.0.4.22  | Enabled                           | GPLv3+      | Sangoma   |
| cdrpro              |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| cel                 | 17.0.2.10  | Enabled                           | GPLv3+      | Sangoma   |
| certman             | 17.0.3.13  | Enabled                           | AGPLv3+     | Sangoma   |
| cidlookup           | 17.0.1.1   | Enabled                           | GPLv3+      | Sangoma   |
| conferences         | 17.0.3.2   | Enabled                           | GPLv3+      | Sangoma   |
| conferencespro      |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| configedit          | 17.0.1.4   | Enabled                           | AGPLv3+     | Sangoma   |
| contactmanager      | 17.0.5.12  | Enabled                           | GPLv3+      | Sangoma   |
| core                | 17.0.17    | Enabled                           | GPLv3+      | Sangoma   |
| cos                 |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| customappsreg       | 17.0.1     | Enabled                           | GPLv3+      | Sangoma   |
| customcontexts      |            | Not Installed (Locally available) | GPLv2+      | Sangoma   |
| dahdiconfig         |            | Not Installed (Locally available) | GPLv3+      | Sangoma   |
| dashboard           | 17.0.4.6   | Enabled                           | AGPLv3+     | Sangoma   |
| daynight            | 17.0.1.1   | Enabled                           | GPLv3+      | Sangoma   |
| dictate             | 17.0.1.2   | Enabled                           | GPLv3+      | Sangoma   |
| directory           | 17.0.1.1   | Enabled                           | GPLv3+      | Sangoma   |
| disa                | 17.0.6     | Enabled                           | AGPLv3+     | Sangoma   |
| donotdisturb        | 17.0.2.3   | Enabled                           | GPLv3+      | Sangoma   |
| dynroute            | 17.0.3.2   | Enabled                           | GPLv3+      | Sangoma   |
| endpoint            | 17.0.1.88  | Enabled                           | Commercial  | Sangoma   |
| extensionroutes     |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| extensionsettings   | 17.0.1     | Enabled                           | GPLv3+      | Sangoma   |
| fax                 | 17.0.3.4   | Enabled                           | GPLv3+      | Sangoma   |
| faxpro              |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| featurecodeadmin    | 17.0.2     | Enabled                           | GPLv3+      | Sangoma   |
| filestore           | 17.0.2.30  | Enabled                           | AGPLv3      | Sangoma   |
| findmefollow        | 17.0.4.10  | Enabled                           | GPLv3+      | Sangoma   |
| firewall            | 17.0.1.28  | Enabled                           | AGPLv3+     | Sangoma   |
| framework           | 17.0.19.16 | Enabled                           | GPLv2+      | Sangoma   |
| hotelwakeup         | 17.0.1.6   | Enabled                           | GPLv2       | Sangoma   |
| iaxsettings         | 17.0.1     | Enabled                           | AGPLv3      | Sangoma   |
| infoservices        | 17.0.1.1   | Enabled                           | GPLv2+      | Sangoma   |
| ivr                 | 17.0.8     | Enabled                           | GPLv3+      | Sangoma   |
| languages           | 17.0.1     | Enabled                           | GPLv3+      | Sangoma   |
| logfiles            | 17.0.3.3   | Enabled                           | GPLv3+      | Sangoma   |
| manager             | 17.0.6     | Enabled                           | GPLv2+      | Sangoma   |
| miscapps            | 17.0.3     | Enabled                           | GPLv3+      | Sangoma   |
| miscdests           | 17.0.1.1   | Enabled                           | GPLv3+      | Sangoma   |
| missedcall          | 17.0.1.2   | Enabled                           | GPLv3+      | Sangoma   |
| music               | 17.0.5     | Enabled                           | GPLv3+      | Sangoma   |
| oracle_connector    |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| outcnam             | 17.0.1     | Enabled                           | GPLv3+      | Sangoma   |
| outroutemsg         | 17.0.1     | Enabled                           | GPLv3+      | Sangoma   |
| paging              | 17.0.3     | Enabled                           | GPLv3+      | Sangoma   |
| pagingpro           |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| parking             | 17.0.2.5   | Enabled                           | GPLv3+      | Sangoma   |
| parkpro             |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| pbxmfa              |            | Not Installed (Locally available) | Commercial+ | Sangoma   |
| phpinfo             | 17.0.1     | Enabled                           | GPLv2+      | Sangoma   |
| pinsets             | 17.0.3.2   | Enabled                           | GPLv3+      | Sangoma   |
| pinsetspro          |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| pm2                 | 17.0.3.3   | Enabled                           | AGPLv3+     | Sangoma   |
| pms                 |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| presencestate       | 17.0.2.4   | Enabled                           | GPLv3+      | Sangoma   |
| printextensions     | 17.0.1.2   | Enabled                           | GPLv3+      | Sangoma   |
| queueprio           | 17.0.1.4   | Enabled                           | GPLv3+      | Sangoma   |
| queues              | 17.0.1.12  | Enabled                           | GPLv2+      | Sangoma   |
| queuestats          |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| qxact_reports       |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| recording_report    |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| recordings          | 17.0.2.3   | Enabled                           | GPLv3+      | Sangoma   |
| restapps            |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| ringgroups          | 17.0.2.6   | Enabled                           | GPLv3+      | Sangoma   |
| sangomaconnect      |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| sangomacrm          |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| sangomartapi        |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| setcid              | 17.0.1.2   | Enabled                           | GPLv3+      | Sangoma   |
| sipsettings         | 17.0.6.9   | Enabled                           | AGPLv3+     | Sangoma   |
| sipstation          |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| sms                 |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| smsplus             |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| soundlang           | 17.0.4.1   | Enabled                           | GPLv3+      | Sangoma   |
| superfecta          | 17.0.3.5   | Enabled                           | GPLv2+      | Sangoma   |
| synologyabb         |            | Not Installed (Locally available) | GPLv3+      | Sangoma   |
| sysadmin            | 17.0.1.98  | Enabled                           | Commercial  | Sangoma   |
| timeconditions      | 17.0.1.18  | Enabled                           | GPLv3+      | Sangoma   |
| tts                 | 17.0.1.1   | Enabled                           | GPLv3+      | Sangoma   |
| ttsengines          | 17.0.1     | Enabled                           | AGPLv3      | Sangoma   |
| ucp                 | 17.0.4.23  | Enabled                           | AGPLv3+     | Sangoma   |
| userman             | 17.0.6.29  | Enabled                           | AGPLv3+     | Sangoma   |
| vmblast             | 17.0.2     | Enabled                           | GPLv3+      | Sangoma   |
| vmnotify            |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| voicemail           | 17.0.5.23  | Enabled                           | GPLv3+      | Sangoma   |
| voicemail_report    |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| voipinnovations     |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| vqplus              |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| weakpasswords       | 17.0.1     | Enabled                           | GPLv3+      | Sangoma   |
| webcallback         |            | Not Installed (Locally available) | Commercial  | Sangoma   |
| webrtc              | 17.0.2.2   | Enabled                           | GPLv3+      | Sangoma   |
+---------------------+------------+-----------------------------------+-------------+-----------+

### Relevant log output

```shell
PDOException (22001)
SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'name' at row 1

Stack frames (5)
4
PDOException
/var/www/html/admin/modules/core/Core.class.php2684
3
PDOStatement execute
/var/www/html/admin/modules/core/Core.class.php2684
2
FreePBX\modules\Core addTrunk
/var/www/html/admin/modules/core/functions.inc.php4459
1
core_trunks_edit
/var/www/html/admin/modules/core/page.trunks.php203
0
include
/var/www/html/admin/config.php658
/var/www/html/admin/modules/core/Core.class.php

                break;
                case "pjsip":
                    $pjsip = $this->getDriver('pjsip');
                    if($pjsip !== false) {
                        if(!empty($_POST["imports"])){
                            $settings = $this->checkPJSIPsettings($settings, $_POST);
                        }
     
                        $settings = array_merge($settings, $_POST);    
     
                        $pjsip->addTrunk($trunknum,$settings);
                    }
                break;
            }
     
            $sql = "REPLACE INTO `trunks`
            (`trunkid`, `name`, `tech`, `outcid`, `keepcid`, `maxchans`, `failscript`, `dialoutprefix`, `channelid`, `usercontext`, `provider`, `disabled`, `continue`)
            VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?)";
            $sth = $this->database->prepare($sql);
            $sth->execute(array(
                $trunknum,
                $name,
                $settings['tech'],
                $settings['outcid'],
                $settings['keepcid'] ?? 'off',
                $settings['maxchans'] ?? '',
                $settings['failtrunk'] ?? '',
                $settings['dialoutprefix'] ?? '',
                $settings['channelid'] ?? '',
                $settings['usercontext'] ?? NULL,
                $settings['provider'] ?? NULL,
                $settings['disabletrunk'] ?? 'off',
                $settings['continue'] ?? 'off'
            ));
     
            if (isset($settings['dialopts']) && $settings['dialopts'] !== false) {
                $this->freepbx->astman->database_put("TRUNK", $trunknum . '/dialopts',$settings['dialopts']);
            } else {
                $this->freepbx->astman->database_del("TRUNK", $trunknum . '/dialopts');
            }

Arguments

    "SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'name' at row 1"

Environment & details:
GET Data
display 	

"trunks"

POST Data
display 	

"trunks"

extdisplay 	

"OUT_1"

action 	

"edittrunk"

tech 	

"pjsip"

provider 	

""

sv_trunk_name 	

"test"

sv_usercontext 	

""

sv_channelid 	

"test"

npanxx 	

""

trunk_name 	

"test-thisisaverylongnameforasiptrunkbutthatisithepoint1235!"

hcid 	

"on"

outcid 	

""

keepcid 	

"off"

maxchans 	

""

dialoutopts_cb 	

"sys"

continue 	

"off"

disabletrunk 	

"off"

failtrunk_enable 	

"0"

prepend_digit 	

array:2 [▶]

pattern_prefix 	

array:2 [▶]

pattern_pass 	

array:2 [▶]

dialoutprefix 	

""

username 	

""

auth_username 	

""

secret 	

""

authentication 	

"outbound"

registration 	

"send"

language 	

""

sip_server 	

"server.no"

sip_server_port 	

""

context 	

"from-pstn"

transport 	

"0.0.0.0-udp"

dtmfmode 	

"auto"

pjsip_line 	

"true"

send_connected_line 	

"no"

auth_rejection_permanent 	

"off"

forbidden_retry_interval 	

"30"

fatal_retry_interval 	

"30"

retry_interval 	

"60"

expiration 	

"3600"

max_retries 	

"10000"

qualify_frequency 	

"60"

outbound_proxy 	

""

user_eq_phone 	

"no"

contact_user 	

""

from_domain 	

""

from_user 	

""

client_uri 	

""

server_uri 	

""

media_address 	

""

aors 	

""

aor_contact 	

""

match 	

""

support_path 	

"no"

t38_udptl 	

"no"

t38_udptl_ec 	

"none"

t38_udptl_nat 	

"no"

t38_udptl_maxdatagram 	

""

fax_detect 	

"no"

trust_rpid 	

"no"

sendrpid 	

"no"

trust_id_outbound 	

"no"

identify_by 	

"default"

inband_progress 	

"no"

direct_media 	

"no"

rewrite_contact 	

"no"

rtp_symmetric 	

"yes"

media_encryption 	

"no"

force_rport 	

"yes"

message_context 	

""

codec 	

array:3 [▶]

Files empty
Cookies
lang 	

"en_US"

searchHide 	

"1"

extensions-all_bs_table_pageList 	

"100"

extensions-all_bs_table_pageNumber 	

"1"

table-trunk-all_bs_table_pageList 	

"25"

table-trunk-all_bs_table_pageNumber 	

"1"

_ga 	

"GA1.1.100275058.1714704886"

PHPSESSID 	

"cl9rs7mdugb6d2hltgqg3cvseo"

Session
langdirection 	

"ltr"

module_name 	

"core"

module_page 	

"trunks"

AMP_user 	

ampuser {#1}

tail_file 	

array:5 [▶]

mod_logfiles_read_log_channels 	

array:8 [▶]

Server/Request Data
SCRIPT_URL 	

"/admin/config.php"

SCRIPT_URI 	

"https://pbx/admin/config.php"

SSLSETUP 	

"true"

HTACCESS 	

"on"

HTTPS 	

"on"

SSL_TLS_SNI 	

"pbx"

HTTP_HOST 	

"pbx"

HTTP_USER_AGENT 	

"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:132.0) Gecko/20100101 Firefox/132.0"

HTTP_ACCEPT 	

"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"

HTTP_ACCEPT_LANGUAGE 	

"en-US,en;q=0.5"

HTTP_ACCEPT_ENCODING 	

"gzip, deflate, br, zstd"

HTTP_REFERER 	

"https://pbx/admin/config.php?display=trunks&tech=pjsip&extdisplay=OUT_1"

CONTENT_TYPE 	

"multipart/form-data; boundary=---------------------------16667384462594648102335370449"

CONTENT_LENGTH 	

"9171"

HTTP_ORIGIN 	

"https://pbx"

HTTP_DNT 	

"1"

HTTP_SEC_GPC 	

"1"

HTTP_CONNECTION 	

"keep-alive"

HTTP_COOKIE 	

"lang=en_US; searchHide=1; extensions-all.bs.table.pageList=100; extensions-all.bs.table.pageNumber=1; table-trunk-all.bs.table.pageList=25; table-trunk-all.bs.t ▶"

HTTP_UPGRADE_INSECURE_REQUESTS 	

"1"

HTTP_SEC_FETCH_DEST 	

"document"

HTTP_SEC_FETCH_MODE 	

"navigate"

HTTP_SEC_FETCH_SITE 	

"same-origin"

HTTP_SEC_FETCH_USER 	

"?1"

HTTP_PRIORITY 	

"u=0, i"

PATH 	

"/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"

SERVER_SIGNATURE 	

""

SERVER_SOFTWARE 	

"Apache"

SERVER_NAME 	

"pbx"

SERVER_ADDR 	

"<pbx ip>"

SERVER_PORT 	

"443"

REMOTE_ADDR 	

"<my ip>"

DOCUMENT_ROOT 	

"/var/www/html"

REQUEST_SCHEME 	

"https"

CONTEXT_PREFIX 	

""

CONTEXT_DOCUMENT_ROOT 	

"/var/www/html"

SERVER_ADMIN 	

"[no address given]"

SCRIPT_FILENAME 	

"/var/www/html/admin/config.php"

REMOTE_PORT 	

"56113"

GATEWAY_INTERFACE 	

"CGI/1.1"

SERVER_PROTOCOL 	

"HTTP/1.1"

REQUEST_METHOD 	

"POST"

QUERY_STRING 	

"display=trunks"

REQUEST_URI 	

"/admin/config.php?display=trunks"

SCRIPT_NAME 	

"/admin/config.php"

PHP_SELF 	

"/admin/config.php"

REQUEST_TIME_FLOAT 	

1731530006.6663

REQUEST_TIME 	

1731530006

Environment Variables empty
Registered Handlers
0. Whoops\Handler\PrettyPageHandler
```

**Comment by kapilgupta01** [comment]
Currently length of the Trunk name field is set to 50. 
`                        <field name="name" type="string" length="50" default=""/>
`

`1406 Data too long for column 'name' ` Error shows you are trying to add really big name, any specific reason?

**Comment by gruubar** [comment]
Let me clarify - I realized why I got the error from the text, but this shouldn't be able to happen. The error log I made here was just a test trunk to reproduce the issue - the name was intentionally long to be sure I recreated the problem for you to see. The original trunk that I discovered this with wasn't that name, it was just descriptive and indicated that the trunk was no longer used, and apparently this was over 50 characters.

I mean, the easy thing is just limit that web UI field to less than 50 characters. If it happened to me randomly it could happen to someone else and potentially delete a production trunk. I'd do it but I'm not a programmer, sorry.
